Manage command to update hathitrust page counts

rlskoeser opened the pull request:
Adds a new manage command to update page counts for non-excerpted HathiTrust works.

Reports on what was done, and creates log entries so an admin will be able to see the change.

I haven't created any unit tests since I'm not sure we will actually want to keep the script in this form - it seems that there may be multiple things that we'll want to do when we updated hathitrust content (rsync, check excerpts, recalculate pages...)

related to #567 

Currently the `count_pages` method saves the model if changed (unless it's an excerpt!); based on the difficulties I had tripping over that here and some other related investigations, I'm planning to refactor that so it does not save and make sure the calling code handles saving when appropriate - but this can be reviewed independent of that refactor.

laurejt commented on ppa/archive/management/commands/update_hathi_pagecounts.py:
```diff
@@ -0,0 +1,87 @@
+    By default, runs on all non-excerpted, public HathiTrust items.
+    """
+
+    help = __doc__
```
Is the class attribute `help` used?

laurejt commented on ppa/archive/management/commands/update_hathi_pagecounts.py:
```diff
@@ -0,0 +1,87 @@
+        )
+
+    def handle(self, *args, **kwargs):
+        verbosity = kwargs.get("verbosity", self.v_normal)
```
```suggestion
        self.verbosity = kwargs.get("verbosity", self.verbosity)
```

laurejt commented on ppa/archive/management/commands/update_hathi_pagecounts.py:
```diff
@@ -0,0 +1,87 @@
+                    stats["unchanged"] += 1
+
+            except storage_exceptions.ObjectNotFoundException:
+                if verbosity >= self.v_normal:
```
```suggestion
                if self.verbosity >= self.v_normal:
```

laurejt commented on ppa/archive/management/commands/update_hathi_pagecounts.py:
```diff
@@ -0,0 +1,87 @@
+                stats["missing_data"] += 1
+
+        # report a summary of what was done
+        if verbosity >= self.v_normal:
```
```suggestion
        if self.verbosity >= self.v_normal:
```

laurejt commented on ppa/archive/management/commands/update_hathi_pagecounts.py:
```diff
@@ -0,0 +1,87 @@
+        if source_ids:
+            hathi_vols = hathi_vols.filter(source_id__in=source_ids)
+
+        stats = defaultdict(int)
```
For clarity, might it be worth fully declaring the desired dict structure upfront?
```suggestion
        stats = {"update":0, "unchanged":0}
```
This would also remove the use of `defaultdict`

laurejt commented on ppa/archive/management/commands/update_hathi_pagecounts.py:
```diff
@@ -0,0 +1,87 @@
+
+        # report a summary of what was done
+        if verbosity >= self.v_normal:
+            self.stdout.write(
+                f"""Volumes with updated page count: {stats['updated']:,}
+    Page count unchanged: {stats['unchanged']:,}
+    Missing pairtree data: {stats['missing_data']:,}"""
+            )
```
I'm unfamiliar with how spacing for logging is handled, but might it be better to use explicit tabs `\t` instead of spaces? This would improve the file formatting and remove the dependency on triple quotes.

laurejt reviewed (changes requested):
See my comments above. Before merging, the verbosity issue should be resolved.

rlskoeser commented:
@laurejt thank you for the careful review. I've incorporated your suggestions & addressed your comments, and also made the change I was planning, so that the `count_pages` method no longer saves the record.
